src/test/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericsConstructorsTest.java

```diff
@@ -0,0 +1,212 @@
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lang.var;
+
+import static org.openrewrite.java.Assertions.*;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+public class UseVarForGenericsConstructorsTest implements RewriteTest {
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new UseVarForGenericsConstructors())
+          .allSources(s -> s.markers(javaVersion(10)));
+    }
+
+    @Nested
+    class NotApplicable {
+        @Test
+        void forEmptyFactoryMethod() {
+            //language=java
+            rewriteRun(
+              version(
+                java(
+                  """
+                      package com.example.app;
+                      
+                      import java.util.List;
+                      
+                      class A {
+                        void m() {
+                            List<String> strs = List.of();
+                        }
+                      }
+                    """),
+                10
+              )
+            );
+        }
+        @Test
+        void withFactoryMethods() {
+            // this one is handled by UseVarForMethodInvocations
+            //language=java
+            rewriteRun(
+              version(
+                java("""
+                package com.example.app;
+                                    
+                import java.util.List;
+                                    
+                class A {
+                  void m() {
+                      List<String> strs = List.of("one", "two");
+                  }
+                }
+                """),
+                10
+              )
+            );
+        }
+        @Test
+        void forEmptyDiamondOperators() {
+            //language=java
+            rewriteRun(
+              version(
+                java(
+                  """
+                      package com.example.app;
+                      
+                      import java.util.List;
+                      import java.util.ArrayList;
+                      
+                      class A {
+                        void m() {
+                            List strs = new ArrayList<>();
+                        }
+                      }
+                    """),
+                10
+              )
+            );
+        }
+        @Test
+        void withDiamondOperatorOnRaw() {
+            //todo check if this may be possible!, We could transform ArrayList into ArrayList<String>
```

Comment:
I've quickly tried to see the changes needed for this, and I think it's ok to leave it as future improvement since it requires quite not so trivial changes..